native-image/microservices/micronaut-hello-rest-maven-layered/pom.xml

```diff
@@ -0,0 +1,254 @@
+                <buildArg>-H:ApplicationLayerInitializedClasses=io.micronaut.inject.annotation.AnnotationMetadataSupport</buildArg>
+                <buildArg>-H:ApplicationLayerInitializedClasses=io.micronaut.core.io.service.MicronautMetaServiceLoaderUtils</buildArg>
+                <buildArg>-H:-UnlockExperimentalVMOptions</buildArg>
+                <buildArg>--initialize-at-run-time=io.netty</buildArg>
+                <buildArg>-J-Xmx6g</buildArg>
+              </buildArgs>
+            </configuration>
+          </plugin>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <configuration>
+              <excludes>
+                <exclude>**/example/micronaut/**</exclude>
+              </excludes>
+              <testExcludes>
+                <exclude>**/example/micronaut/**</exclude>
+              </testExcludes>
+            </configuration>
+          </plugin>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-install-plugin</artifactId>
+            <executions>
+              <execution>
+                <phase>install</phase>
+                <goals>
+                  <goal>install-file</goal>
+                </goals>
+                <configuration>
+                  <groupId>${project.groupId}</groupId>
+                  <file>${project.build.directory}/${project.artifactId}-${project.version}.jar</file>
+                  <artifactId>micronaut-base-layer</artifactId>
+                  <packaging>jar</packaging>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+    <profile>
+      <id>app-layer</id>
+      <dependencies>
+        <dependency>
+          <groupId>${project.groupId}</groupId>
+          <artifactId>micronaut-base-layer</artifactId>
+          <version>${project.version}</version>
+        </dependency>
+      </dependencies>
+      <build>
+        <directory>${project.basedir}/app-layer-target</directory>
+        <plugins>
+          <plugin>
+            <groupId>org.graalvm.buildtools</groupId>
+            <artifactId>native-maven-plugin</artifactId>
+            <version>0.10.3</version>
+            <configuration>
+              <imageName>layered-app</imageName>
+              <mainClass>example.micronaut.Application</mainClass>
+              <buildArgs>
+                <buildArg>-H:+UnlockExperimentalVMOptions</buildArg>
+                <buildArg>-H:LayerUse=base-layer-target/base-layer.nil</buildArg>
+                <buildArg>-H:LinkerRPath=$ORIGIN</buildArg>
+                <buildArg>-H:-UnlockExperimentalVMOptions</buildArg>
+                <buildArg>--initialize-at-run-time=io.netty</buildArg>
+                <buildArg>-J-Xmx4g</buildArg>
```

Comment:
Again, why set max heap? To show that the app layer needs less memory to build? It's also missing in the README.md.